src/element.js

```diff
@@ -164,29 +164,51 @@ function measureLabelSize(ctx, lines, fonts) {
   return widthCache.get(mapKey);
 }
 
+function toFonts(fonts, fitRatio) {
+  return fonts.map(function(f) {
+    f.size = Math.floor(f.size * fitRatio);
+    f.lineHeight = undefined;
+    return toFont(f);
+  });
+}
+
 function labelToDraw(ctx, rect, options, labelSize) {
   const {overflow, padding} = options;
   const {width, height} = labelSize;
   if (overflow === 'hidden') {
     return !((width + padding * 2) > rect.w || (height + padding * 2) > rect.h);
+  } else if (overflow === 'fit') {
+    const ratio = Math.min(rect.w / (width + padding * 2), rect.h / (height + padding * 2));
+    if (ratio < 1) {
+      return ratio;
+    }
   }
   return true;
 }
 
+function getFontFromOptions(rect, labels) {
+  const {font, hoverFont} = labels;
+  const optFont = (rect.active ? hoverFont : font) || font;
```

Comment:
I think I'm going in circles, but again was it so that the sub-objects (labels in this case) don't automatically resolve the hover options?

Reply:
Yes @kurkle 
There was also an issue that we closed with a patch. Hover font seems not solved. But tomorrow I want to understand better the current logic in chartjs and why is not applied here for font. Ok for you?

Reply:
@kurkle please, don't hate me if you can... when you have time, we should go ahead with PRs on the annotation, I think...